app/api/route/user_route.py

```diff
@@ -22,21 +23,21 @@
 @user_blueprint.input(user_schemas.UserSchema)
 @user_blueprint.output(user_schemas.UserSchema, status_code=201)
 @user_blueprint.auth_required(api_key_auth)
-def user_post(user_params: users.CreateUserParams) -> dict:
+@flask_db.with_db_session_transaction
+def user_post(db_session: db.Session, user_params: users.CreateUserParams) -> dict:
```

Comment:
I definitely like this approach for the DB session with the APIs. Since we already have half a dozen decorators, it's clear that a lot is being injected into this method already. Since this method is already an entry-point, I think having the session there makes a lot of sense.